--- ./funcoeszz.orig	2011-02-03 17:24:25.000000000 -0200
+++ ./funcoeszz	2011-02-03 17:29:10.000000000 -0200
@@ -6546,7 +6546,7 @@ zzsenha ()
 
 	local posicao letra
 	local n=6
-	local alpha='abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ0123456789'
+	local alpha='abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ0123456789!@#$%&*(){}[]/,.":?~^-|'
 	local maximo=${#alpha}
 
 	# Guarda o número informado pelo usuário (se existente)
@@ -6574,9 +6574,9 @@ zzsenha ()
 	do
 		n=$((n-1))
 		posicao=$((RANDOM % ${#alpha} + 1))
-		letra=$(echo -n "$alpha" | cut -c$posicao)
-		alpha=$(echo $alpha | tr -d $letra)
-		echo -n $letra
+		letra="$(echo -n "$alpha" | cut -c$posicao)"
+		alpha=$(echo "$alpha" | tr -d "$letra")
+		echo -n "$letra"
 	done
 	echo
 }
